### PR TITLE
Support for Team Drive

### PR DESCRIPTION
**Pull request by x1ddos**
See details at the API docs page:
https://developers.google.com/drive/v3/web/enable-teamdrives

This also adds a nicer error message. For instance, instead of 403
or 404 code, the program displays the actual error message
as sent from Drive.

R: @marcacohen

### CODE REVIEW

File: docshare.go [hunk 1]
```diff
@@ -89,59 +96,115 @@ func main() {
 			ch <- struct{}{}
 			wg.Add(1)
 			go func(docID string) {
+				// TODO: figure out a better way to create context for each goroutine.
+				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```
Comment (mco-gh) [inline docshare.go:100]: Why worry about timeouts only for remove but not add?

**Review by mco-gh — APPROVED** [review]
LGTM, thanks for cleaning this up. I just wondered about the asymmetry of timeout context.

File: docshare.go (hunk 1 above)
Comment (x1ddos) [reply docshare.go:100]: It's just I always try to make minimal changes. In this case I only needed the new function for removal, which I thought would be better with context. The "add" part only needed some URL parameters, so adding context would've been out of scope for this change 

In fact, the PR #3 does add context to the other part.

Comment (x1ddos) [reply docshare.go:100]: I have to admin though, changing log message is also out of scope for this PR. I guess I should've added it separately but here we are.

**Pull request merged**